src/utils/referral.js

```diff
@@ -255,7 +256,7 @@ const extractReasonCode = (data) => {
     ),
   ].filter((elem) => elem !== `none`);
   reasons.forEach((reason) => {
-    const coding = echisNHDDValuesCoding[reason];
+    const coding = !!service ? serviceMapping[service].mapping[reason] : echisNHDDValuesCoding[reason];
```

Comment:
Could you use Typescripts syntax

e.g `serviceMapping[service]?.mapping[reason]` ?